geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/index/z3/package.scala

```diff
@@ -25,17 +25,21 @@ package object z3 {
     }
   }
 
-  case class Z3IndexValues(sfc: Z3SFC,
-                           geometries: FilterValues[Geometry],
-                           spatialBounds: Seq[(Double, Double, Double, Double)],
-                           intervals: FilterValues[Bounds[ZonedDateTime]],
-                           temporalBounds: Map[Short, Seq[(Long, Long)]],
-                           temporalUnbounded: Seq[(Short, Short)])
+  case class Z3IndexValues(
+      sfc: Z3SFC,
+      geometries: FilterValues[Geometry],
+      spatialBounds: Seq[(Double, Double, Double, Double)],
+      intervals: FilterValues[Bounds[ZonedDateTime]],
+      temporalBounds: Map[Short, Seq[(Long, Long)]],
+      temporalUnbounded: Seq[(Short, Short)]
+    ) extends TemporalIndexValues
 
-  case class XZ3IndexValues(sfc: XZ3SFC,
-                            geometries: FilterValues[Geometry],
-                            spatialBounds: Seq[(Double, Double, Double, Double)],
-                            intervals: FilterValues[Bounds[ZonedDateTime]],
-                            temporalBounds: Map[Short, (Double, Double)],
-                            temporalUnbounded: Seq[(Short, Short)])
+  case class XZ3IndexValues(
+      sfc: XZ3SFC,
+      geometries: FilterValues[Geometry],
+      spatialBounds: Seq[(Double, Double, Double, Double)],
+      intervals: FilterValues[Bounds[ZonedDateTime]],
+      temporalBounds: Map[Short, (Double, Double)],
+      temporalUnbounded: Seq[(Short, Short)]
```

Comment:
What does "temporalUnbounded" mean?

Reply:
those are epochs that don't have any temporal endpoints, e.g. if you query `dtg > today` you'd end up with `(today's epoch, Short.MaxValue)`. that way we can scan just with a start (or end) row, instead of trying to calculate z-values for every single epoch. this is mainly used for secondary attribute index scans, and was added in https://geomesa.atlassian.net/browse/GEOMESA-2402

Reply:
i'll add some scala docs